Sources/TensorFlow/Optimizer.swift

```diff
@@ -355,3 +355,102 @@ public class AdaGrad<Model: Layer>: Optimizer
         update(&model.allDifferentiableVariables, along: direction)
     }
 }
+
+/// Adadelta Optimizer
+///
+/// It is a method that uses the magnitude of recent gradients 
+/// and steps to obtain an adaptive step rate. 
+/// An exponential moving average over the gradients and steps is kept;
+/// a scale of the learning rate is then obtained by their ration.
+/// 
+/// Reference: ["ADADELTA: An Adaptive Learning Rate Method"](
+/// https://arxiv.org/abs/1212.5701)
+///
+public class AdaDelta<Model: Layer>: Optimizer
+    where Model.AllDifferentiableVariables == Model.TangentVector {
+    public typealias Model = Model
+    /// The learning rate.
+    public var learningRate: Float
+    ///decay factor, corresponding to fraction of gradient to keep at each time step.
+    public var rho: Float
+    /// A small scalar added to the denominator to improve numerical stability.
+    public var epsilon: Float
+    /// The learning rate decay.
+    public var decay: Float
+    /// The current step.
+    public var step: Int = 0
+    /// Accumulate Gradients
```

Comment:
Two suggestions on documentation comments:
- Properties should be described with a noun phrase ending with a period.
- Documentation comments for mathematical variables should use the term of art. In this case, we should switch to the accurate, proper terminology that's used by the paper.
```suggestion
    /// The accumulated, exponentially decaying average of squared gradients.
```